tokio/src/task/local.rs

```diff
@@ -695,4 +779,28 @@ impl task::Schedule for Arc<Shared> {
     fn schedule(&self, task: task::Notified<Self>) {
         Shared::schedule(self, task);
     }
+
+    cfg_unstable! {
+        fn unhandled_panic(&self) {
+            use crate::runtime::UnhandledPanic;
+
+            match self.unhandled_panic {
+                UnhandledPanic::Ignore => {
+                    // Do nothing
+                }
+                UnhandledPanic::ShutdownRuntime => {
+                    // This hook is only called from within the runtime, so
+                    // `CURRENT` should match with `&self`, i.e. there is no
+                    // opportunity for a nested scheduler to be called.
+                    CURRENT.with(|maybe_cx| match maybe_cx {
+                        Some(cx) if Arc::ptr_eq(self, &cx.shared) => {
+                            cx.unhandled_panic.set(true);
+                            cx.owned.close_and_shutdown_all();
+                        }
+                        _ => panic!("runtime core not set in CURRENT thread-local"),
```

Comment:
nit: should probably be
```suggestion
                        _ => unreachable!("runtime core not set in CURRENT thread-local"),
```